shell/common/node_bindings.cc

```diff
@@ -415,15 +418,19 @@ void NodeBindings::Initialize() {
 
 node::Environment* NodeBindings::CreateEnvironment(
     v8::Handle<v8::Context> context,
-    node::MultiIsolatePlatform* platform) {
+    node::MultiIsolatePlatform* platform,
+    std::vector<std::string> args,
+    std::vector<std::string> exec_args) {
+  if (browser_env_ != BrowserEnvironment::kUtility) {
 #if BUILDFLAG(IS_WIN)
-  auto& atom_args = ElectronCommandLine::argv();
-  std::vector<std::string> args(atom_args.size());
-  std::transform(atom_args.cbegin(), atom_args.cend(), args.begin(),
-                 [](auto& a) { return base::WideToUTF8(a); });
+    auto& atom_args = ElectronCommandLine::argv();
+    args.resize(atom_args.size());
+    std::transform(atom_args.cbegin(), atom_args.cend(), args.begin(),
+                   [](auto& a) { return base::WideToUTF8(a); });
 #else
-  auto args = ElectronCommandLine::argv();
+    args = ElectronCommandLine::argv();
 #endif
+  }
```

Comment:
nit: let's refactor this to be less error-prone. Have two versions of `NodeBindings::CreateEnvironment`, one which takes `args` and `exec_args`, and one which doesn't. The version that doesn't take extra args should contain this logic for pulling the args from globals, then pass those into the version that takes `args`.